src-graphql/client.js

```diff
@@ -24,11 +24,11 @@ function fetchAllProductResources(product, client) {
   const images = product.images;
   const variants = product.variants;
 
-  if (images && images[images.length - 1].hasNextPage.valueOf()) {
+  if (images && images.length && images[images.length - 1].hasNextPage.valueOf()) {
```

Comment:
or do you mean to attach page info back to the set, and not load it onto every node?